js/ui/layout.js

```diff
@@ -421,6 +424,22 @@ const LayoutManager = new Lang.Class({
                                 Lang.bind(this, this._updateHotCorners));
 
         this._monitorsChanged();
+
+        /*
+         * When we're not in the Overview and the text scaling factor changes, the
+         * Overview icons don't update to match the new font size. To fix that, and
+         * some random font rendering issues in apps, toggle the Overview.
+         */
+        Main.interfaceSettings.connect('changed::' + KEY_TEXT_SCALING_FACTOR, Lang.bind(this, function() {
+            if (!this._inOverview) {
+                Main.layoutManager.showOverview();
+
+                GLib.idle_add(GLib.PRIORITY_DEFAULT, Lang.bind(this, function() {
+                                  this.hideOverview();
+                                  return false;
+                              }));
```

Comment:
Hmmm, showing the overview and then hiding it again immediately on a setting change seems suspect to me. Can't we just apply this setting change once the overview is next toggled?